src/zarr/core/group.py

```diff
@@ -704,29 +705,29 @@ async def arrays(self) -> AsyncGenerator[AsyncArray, None]:
     async def tree(self, expand: bool = False, level: int | None = None) -> Any:
         raise NotImplementedError
 
-    async def empty(self, **kwargs: Any) -> AsyncArray:
-        raise NotImplementedError
+    async def empty(self, shape: ChunkCoords, **kwargs: Any) -> AsyncArray:
+        return await async_api.empty(shape=shape, **kwargs)
```

Comment:
I'm realizing that whoever wrote these stub method signatures was not really thinking (it was me 🤦 ).

The 2.x signature was this: 

```python
def empty(self, name: str, **kwargs) -> Array:
```
Numpy's equivalent is:
```python
numpy.empty(shape, dtype=float, order='C', *, device=None, like=None)
```

Here's a proposed signature with some pseudo code:

```suggestion
    async def empty(self, *, shape: ChunkCoords, name: str, **kwargs: Any) -> AsyncArray:
        path = self.store_path / name
        return await async_api.empty(shape=shape, path=path**kwargs)
```

Reply:
I am wondering if we want to set `path=self.store_path / name` here. It seems like in `create()` we append the `path` to the `store`: https://github.com/zarr-developers/zarr-python/blob/v3/src/zarr/api/asynchronous.py#L729-L730. Would we want the `path` to just be the `name` then?